### PR TITLE
rustdoc: adjust spacing and typography in header

### LINKED ISSUE #131589: crate name is too small
for example: 

when read https://docs.rs/futures/latest/futures/

Image

the crate name size is big, as the crate **name**, this size is ok to hint people "the crate name is XXX"

however, when enter a module sub page, the crate name become too small

https://docs.rs/futures/latest/futures/future/index.html

Image

it is too small, even make me wonder if my browser is broken to render the website, so I change to firefox, but nothing changed

also we can see the up spacing is shorter than the lower spacing, tha't so strange

Image

and the **source** up spacing is short, but the lower spacing is too long, even three times the length

Image

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #131589

Preview: https://notriddle.com/rustdoc-html-demo-12/spacing/std/index.html

| Before | After |
|--|--|
| image | image
| image | image
| image | image
| image | image

First of all, we put 4px additional margin below the search box, and 4px margin below the header to balance it out.

The bigger problem we have to solve is making the lines look logically spaced. This is troublesome, because Fira Sans (the typeface we use here) wants to look good on average, and to avoid breaking, with text that uses [ascenders and descenders](https://www.w3.org/TR/css-inline-3/images/text-edge.png). If the text we're putting in happens to not have any, things look weird (strictly speaking, there’s hand-tuning here, because the Copy Path button messes with stuff, but the overall point is that there is no true, one perfect layout).

In order to play nicely with the font, I've tweaked the text to use that space. The word "Source" for the link is now capitalized, and the Since version number now uses oldstyle nums with descenders.

### CODE REVIEW

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -164,6 +164,9 @@ h1, h2, h3, h4 {
 .docblock > h6:first-child {
 	margin-top: 0;
 }
+.since, .version {
+	font-variant-numeric: oldstyle-nums;
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/css/rustdoc.css:168]: What's the reason behind this change?

Comment (notriddle) [reply src/librustdoc/html/static/css/rustdoc.css:168]: In the below screenshot of the new version, the three red lines are the same length. As you can see, the digits are the same height as the lowercase letters.

377982871-80f450af-ccc8-46d4-82ea-d4ea428f9276

In the below screenshot of the old version, the red lines are also the same length. As you can see, the digits are taller than the lowercase letters.

377982837-b5c5132d-1e5e-402e-ba19-1dea9e70ea6f

The problem with the old version is that adding the digits makes the subheader appear closer to the main header, even if its CSS box doesn't move. This means that, in the old version, there is no "correct" spacing that will look right both with and without the digits.

Comment (notriddle) [reply src/librustdoc/html/static/css/rustdoc.css:168]: I don't feel that strongly about it. If you want to stick with the present-day format, it's okay.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks for the improvement!

@bors r+ rollup
